src/main/java/spoon/SpoonModelBuilder.java

```diff
@@ -76,22 +76,24 @@
 	 */
 	boolean build(JDTBuilder builder);
 
+	/** The types of compilable elements */
+	enum InputType {
+		FILES, CTTYPES
+	}
+
+
 	/**
 	 * Generates the bytecode associated to the classes stored in this
 	 * compiler's factory. The bytecode is generated in the directory given by
 	 * {@link #getBinaryOutputDirectory()}.
 	 *
-	 * @see #getSourceClasspath()
-	 */
-	boolean compile();
```

Comment:
`SpoonModelBuilder` is an interface (and very exposed). Think to deprecate methods rather than remove them like this.